server/bundler/remove-parent-modules-monkeypatch.js

```diff
@@ -0,0 +1,116 @@
+// this is really gross.
+//
+// originally from
+/*
+	MIT License http://www.opensource.org/licenses/mit-license.php
+	Author Tobias Koppers @sokra
+*/
+"use strict"; //eslint-disable-line
+
+var forEach = require( 'lodash/forEach' ),
+	filter = require( 'lodash/filter' ),
+	indexOf = require( 'lodash/indexOf' ),
+	pullAll = require( 'lodash/pullAll' ),
+	nanotimer = require( 'nanotimer' );
+
+var RemoveParentModulesPlugin = require( 'webpack/lib/optimize/RemoveParentModulesPlugin' );
+var Chunk = require( 'webpack/lib/Chunk' );
+var Module = require( 'webpack/lib/Module' );
+
+module.exports = RemoveParentModulesPlugin;
+
```

Comment:
this is a direct lift of the normal RemoveParentModulesPlugin, rewritten to use a Set to track checked chunks instead of an array.

The thing is, we never have _that many_ chunks, so the O(1) lookup on a Set doesn't really help vs O(n) lookup on the array-as-set webpack uses. We just don't have that many chunks.